Added Usage chapter to README file (#12)
openshift-kni/eco-goinfra: README.md

### Project requirements
* golang v1.19.x

## Usage
In order to re-use code from this project you need to import relevant package/packages in to your project code.

```go
import "github.com/openshift-kni/eco-goinfra/pkg/NAME_OF_A_NEEDED_PACKAGE"
```

### Clients package:

> [clients](./pkg/clients) package contains several api clients combined in to the single struct.

Reviewer: ```suggestion
The [clients](./pkg/clients) package contains several api clients combined into a single struct.
```

> The New function of client package returns ready connection to cluster api.

Reviewer: ```suggestion
The New function of the clients package returns a ready connection to the cluster api.
```

If the path to kubeconfig is not specified to the new function then the KUBECONFIG environment variable is used.
In case of failure client.New("") returns nil.
```go
import "github.com/openshift-kni/eco-goinfra/pkg/clients"

func main() {
    apiClients := clients.New("")
	
    if apiClients == nil {
        panic("Failed to load api client")
        }
    }
)
```
[Client usage example](./usage/client/client.go)

### Cluster Objects
Every cluster object namespace, configmap, daemonset, deployment and other has its own package under [packages](./pkg) directory.
The structure of any object has common interface:
```go

> func NewBuilder() or New[ObjectName]Builder() // Initiates object struct. This function require minimum set of parameters that allow to create object on a cluster.

Reviewer: ```suggestion
func NewBuilder() or New[ObjectName]Builder() // Initiates object struct. This function require minimum set of parameters that are required to create the object on a cluster.
```
